rust/src/storage/file/mod.rs

```diff
@@ -85,84 +84,52 @@ impl StorageBackend for FileStorageBackend {
     }
 
     async fn put_obj(&self, path: &str, obj_bytes: &[u8]) -> Result<(), StorageError> {
-        let tmp_path = create_tmp_file_with_retry(self, path, obj_bytes).await?;
-
-        if let Err(e) = rename::atomic_rename(&tmp_path, path) {
-            fs::remove_file(tmp_path).await.unwrap_or(());
-            return Err(e);
+        if let Some(parent) = Path::new(path).parent() {
+            fs::create_dir_all(parent).await?;
         }
-
-        Ok(())
-    }
-}
-
-const CREATE_TEMPORARY_FILE_MAX_TRIES: i32 = 5;
-
-async fn create_tmp_file_with_retry(
-    backend: &FileStorageBackend,
-    path: &str,
-    obj_bytes: &[u8],
-) -> Result<String, StorageError> {
```

Comment:
there's no retry with these changes anymore since they didn't suite well with new approach, we might reconsider it's later